### PR TITLE
fix: broken shutdown for services on fraud proofs receival 

### LINKED ISSUE #1205: nodebuilder: broken shutdown for services on fraud proofs recieval
The problem comes after recent node refactoring and is located in [using start context](https://github.com/celestiaorg/celestia-node/blob/main/nodebuilder/header/module.go#L61-L62) as lifecycle context. The start context is short-lived and is canceled after the node is started. The lifecycle context is long-lived and needed there to ensure that the goroutine spawned inside `fraud.Lifecycle` is cleaned up when the node stops. Right now, the goroutine is cleaned up right after the node starts, preventing listening for any fraud proofs. 

The real issue is the lifecycle context pattern being rather obscure, but that's a discussion for another day, and the quick solution will likely get the right context and set it in the right place.

### PR DESCRIPTION
**Pull request by distractedm1nd**
Closes #1205

### CODE REVIEW

File: nodebuilder/header/module.go [hunk 1]
```diff
@@ -43,7 +43,9 @@ func ConstructModule(tp node.Type, cfg *Config) fx.Option {
 		}),
 		fx.Provide(fx.Annotate(
 			newSyncer,
-			fx.OnStart(func(ctx context.Context, lc fx.Lifecycle, fservice fraudServ.Module, syncer *sync.Syncer) error {
+			// we skip the first passed context which only lives for the OnStart operation,
+			// because fraudServ.Lifecycle needs a long living context.
+			fx.OnStart(func(_, ctx context.Context, lc fx.Lifecycle, fservice fraudServ.Module, syncer *sync.Syncer) error {
```
Comment (Wondertan) [inline nodebuilder/header/module.go:48]: Both contexts should be used.

**Review by Wondertan — COMMENTED** [review]
This should be fixed in every place where fraud.Lifecycle is used

**Review by Wondertan — COMMENTED** [review]
`modfraud.Lifecycle` is also used in DASer, State

**Review by Wondertan — COMMENTED** [review]
Verifed locally and LGTM

File: nodebuilder/daser/module.go [hunk 2]
```diff
@@ -19,9 +19,9 @@ func ConstructModule(tp node.Type) fx.Option {
 			"daser",
 			fx.Provide(fx.Annotate(
 				NewDASer,
-				fx.OnStart(func(ctx context.Context, lc fx.Lifecycle, fservice fraudServ.Module, das *das.DASer) error {
+				fx.OnStart(func(startCtx, ctx context.Context, lc fx.Lifecycle, fservice fraudServ.Module, das *das.DASer) error {
 					lifecycleCtx := fxutil.WithLifecycle(ctx, lc)
```
[the remark this answers is not in the document]
Comment (distractedm1nd) [reply nodebuilder/daser/module.go:23]: do you mean wrapping it in nodebuilders root?

[the remark this answers is not in the document]
Comment (Wondertan) [reply nodebuilder/daser/module.go:23]: Yes

[the remark this answers is not in the document]
Comment (distractedm1nd) [reply nodebuilder/daser/module.go]: done

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
The FraudSyncing test was failing because of https://github.com/celestiaorg/celestia-node/issues/1236. Making the explicit connection light-> full fixed the issue

File: state/core_access.go [hunk 3]
```diff
@@ -98,7 +98,8 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 func (ca *CoreAccessor) Stop(context.Context) error {
 	defer ca.cancel()
 	if ca.coreConn == nil {
-		return fmt.Errorf("core-access: no connection found to close")
+		log.Warn("no connection found to close")
+		return nil
```
Comment (renaynay) [inline state/core_access.go:102]: Good.

**Review by renaynay — APPROVED** [review]

**Pull request merged**